protocol/test/elder-election-tests.ts

```diff
@@ -109,92 +112,114 @@ describe("Elder Election", async () => {
-//           deadline,
-//           nonce,
-//         ]
-//       );
-//       const messageBytes = ethers.utils.arrayify(message);
-//       const signature = await voter.signMessage(messageBytes)
-
-//       return {
-//         account,
-//         deadline,
-//         discordIds,
-//         signature,
-//       }
-//     } 
-
-//     {
-//       const req = await signedSetEndorsementReq(ben, [DISCORD_ID_1]);
-//       await expect(ELDER_ELECTION.relayedSetEndorsementsFor(
-//         req.account,
-//         req.discordIds,
-//         req.deadline - 600,
-//         req.signature,
-//       )).to.be.revertedWith("DeadlineExpired");
-//     }
-
-//     {
-//       const req = await signedSetEndorsementReq(ben, [DISCORD_ID_1]);
-//       await expect(ELDER_ELECTION.relayedSetEndorsementsFor(
-//         req.account,
-//         req.discordIds,
-//         req.deadline,
-//         await ben.signMessage("XXXX"),
-//       )).to.emit(ELDER_ELECTION, "InvalidSignature");
-//     }
-
-//     {
-//       const req = await signedSetEndorsementReq(ben, [DISCORD_ID_1]);
-//       await expect(ELDER_ELECTION.relayedSetEndorsementsFor(
-//         req.account,
-//         req.discordIds,
-//         req.deadline,
-//         req.signature,
-//       )).to.emit(ELDER_ELECTION, "UpdateEndorsements");
-//       }
-//   });
+  it("relayed endorsements work", async () => {
+
+    const provider = ethers.getDefaultProvider();
+    const chainId = 31337;
+
+    {
+      const election = ELDER_ELECTION.connect(nominator);
+      await election.nominate(DISCORD_ID_1);
+      await election.nominate(DISCORD_ID_2);
+    }
+
+    async function signedReq(voter: MySigner, discordIds: number[]):
+       Promise<{req: ElderElection.EndorsementReqStruct, signature: Uint8Array}> {
+
+      const block = await provider.getBlock(await provider.getBlockNumber());
+      const now = block.timestamp;
+
+      const account = await voter.getAddress();
+      const deadline = now + 120;
+      const nonce = await ELDER_ELECTION.nonces(account);
+
+      const req: ElderElection.EndorsementReqStruct = {
+        account,
+        discordIds,
+        deadline: deadline,
+        nonce: nonce,
+      };
+
+      const domain: TypedDataDomain = {
+        name: 'ElderElection',
+        version: '1',
+        chainId,
+      };
+
+      const types: Record<string, TypedDataField[]> = {
+        EndorsementReq: [
+          { name: 'account', type: 'address' },
+          { name: 'discordIds', type: 'uint256[]' },
+          { name: 'deadline', type: 'uint256' },
+          { name: 'nonce', type: 'uint256' },
+
+        ]
+      };
+
+      const value: Record<string, unknown> = {
+        account: req.account,
+        discordIds: req.discordIds,
+        deadline: deadline,
+        nonce: nonce,
+      };
+
+      const signature = await voter._signTypedData(domain, types, value);
+
+      return {
+        req,
+        signature: ethers.utils.arrayify(signature),
```

Comment:
what does arrayifying the sig do? don't think I've used this before

Reply:
in this case it turns a string into a Uint8Array.